app/controllers/pieces_controller.rb

```diff
@@ -0,0 +1,18 @@
+class PiecesController < ApplicationController
+  def index
+    game_id = request.referrer.last.to_i
```

Comment:
You can remove the referrer logic by nesting the pieces controller under the game.  In `routes.rb`:

```ruby
resources :game do
  resources :pieces, only: [:index, :update]
end
```

Then in your controller actions here, you will have a `param[:game_id]` available to fetch the game.  The URLs will change to `/games/:game_id/pieces` and `/games/:game_id/pieces/:id`.  You don't have to change the name or location of the PiecesController